### ISSUE TITLE
Add diffusion informations to extended metadata

### ISSUE THREAD

**Issue by yblatti** [easySDI/easySDI#740]
**Original author :** Blatti Yves

We want to display some informations about diffusion in our search results and in metadata form. For example:
- Properties
- Maximum area for order
(See screenshot)

For this we need to add some infos in "ExtendedXML":https://forge.easysdi.org/projects/easysdi/wiki/4_catalog_configuration#Extended-XML

**Fixed versions :** 4.3.0

**Comment by yblatti** [comment]
**Original author :** Blatti Yves

I propose to add the following in Extended XML:
* In the diffusion node: minimal and maximal surface for order (if any)
* A child node for properties. Properties with attributes : id, name, alias, type, mandatory and translated label
** For each property, it's values with attributes : id, name, alias and translated label
** Properties values are included for the next property types:
*** LIST
*** MULTIPLELIST
*** CHECKBOX
*** MESSAGE
*** (there's no value to display for TEXT and TEXTAREA)


Sample:

```

<code class="xml">
<sdi:ExtendedMetadata lang="association" callfromjoomla="1">
        <sdi:ex_Resource name="aaaaaaaaaaaaa" descriptionLength="">
            <sdi:ex_Metadata created="2014-07-02 12:49:57" updated="2014-07-02 12:49:57">
                <sdi:ex_Diffusion isfree="true" isDownladable="false" isOrderable="true"
                                  surfacemin="" surfacemax="100000000000000000000000" 
                                  file_size="" size_unit="" file_type="">
                    <sdi:ex_ExtractionProperties>
                        <sdi:ex_ExtractionProperty id="1" name="Format" alias="FORMAT"
                                                   label="Format" 
                                                   mandatory="true" type="list">
                            <sdi:ex_PropertyValue id="6" name="BMBLT" alias="bmblt" label="BMBLT">
                            </sdi:ex_PropertyValue>
                            <sdi:ex_PropertyValue id="10" name="DWG" alias="dwg" label="DWG">
                            </sdi:ex_PropertyValue>
                            <sdi:ex_PropertyValue id="39" name="DXF Geobat MD.01" alias="dxf-geobat-md01" label="DXF Geobat MD.01">
                            </sdi:ex_PropertyValue>
                        </sdi:ex_ExtractionProperty>
                        <sdi:ex_ExtractionProperty id="2" name="Système d'exploitation" alias="os"
                                                   label="Système d'exploitation"
                                                   mandatory="true" type="list">
                            <sdi:ex_PropertyValue id="17" name="MAC" alias="mac" label="MAC">
                            </sdi:ex_PropertyValue>
                            <sdi:ex_PropertyValue id="19" name="WINDOWS" alias="windows" label="WINDOWS">
                            </sdi:ex_PropertyValue>
                        </sdi:ex_ExtractionProperty>
                    </sdi:ex_ExtractionProperties>
                </sdi:ex_Diffusion>
            </sdi:ex_Metadata>
            <sdi:ex_Version name="2014-07-02 12:49:57">
            </sdi:ex_Version>
            <sdi:ex_Resourcetype name="Géoproduit" alias="geoproduit">
                <sdi:ex_Logo path="" width="" height="">
                </sdi:ex_Logo>
            </sdi:ex_Resourcetype>
            <sdi:ex_Organism name="ASIT-VD">
                <sdi:ex_Logo path="http://www.asitvd.ch/images/joomla_logo_black.jpg" width="" height="">
                </sdi:ex_Logo>
            </sdi:ex_Organism>
        </sdi:ex_Resource>
        <sdi:links>
            <sdi:parents>
            </sdi:parents>
            <sdi:children>
            </sdi:children>
        </sdi:links>
        <sdi:applications>
        </sdi:applications>
        <sdi:action>
            <sdi:extraction>
                <sdi:html>
                </sdi:html>
            </sdi:extraction>
            <sdi:sheetview>
                <sdi:link>http://dev.t907507.adacv.ad.etat-de-vaud.ch/index.php?option=com_easysdi_catalog&amp;view=sheet&amp;guid=cd4d69ab-e0a0-3f64-2121-5bfae7eb7069&amp;lang=association&amp;catalog=Main&amp;preview=search_list&amp;type=</sdi:link>
            </sdi:sheetview>
            <sdi:exportpdf>
                <sdi:link>http://dev.t907507.adacv.ad.etat-de-vaud.ch/index.php?option=com_easysdi_catalog&amp;task=sheet.exportPDF&amp;id=cd4d69ab-e0a0-3f64-2121-5bfae7eb7069&amp;lang=association&amp;catalog=Main&amp;preview=search_list&amp;type=</sdi:link>
            </sdi:exportpdf>
            <sdi:exportxml>
                <sdi:link>http://dev.t907507.adacv.ad.etat-de-vaud.ch/index.php?option=com_easysdi_catalog&amp;task=sheet.exportXML&amp;id=cd4d69ab-e0a0-3f64-2121-5bfae7eb7069</sdi:link>
            </sdi:exportxml>
        </sdi:action>
    </sdi:ExtendedMetadata>

```

**Comment by yblatti** [comment]
**Original author :** Blatti Yves

2 weeks without comments, what do we do?

**Comment by yblatti** [comment]
**Original author :** Magoni Bruno

It's summer holidays ;-)

Shopping properties can logically be added inside the metadata but I see some relevant points:
* shopping properties are multilingual so label should managed multilingual entries; when adding a new translation language in easySDI CATALOG options, what about updating all metadata content with such labels; I guess that XML attributes "name" and "label" shouldn't be integrated into extended metadata content and that XLST could do the trick (multilingual metadata content is already driven with XSLT).
* shopping properties can be managed from backoffice interface (add, remove, update) so it's important to display message to the Administrator that any properties changes can affect XSLT render...

Such points are also applicable to already existing extended content such like ResourceType's name, Organism's name, Parent resource's name

As such easySDI extended content is growing more and more, should we think about how publishing it with ISO19115 compliance (Metadata extension information package) and how publishing or not such extended content (CSW Proxy "easySDI vendor specific" parameter) ?

**Comment by yblatti** [comment]
**Original author :** Blatti Yves

Hi!

+For point 1: Multi language:+ 
ExtendedMetadata is not stored in OGC catalog (GeoNetwork), but generated on request. So the content is generated with user language.
(in this proposal labels of property and propertyvalue, are the only translated items)
Having name and label make thing easier in XSL, and if we add a new value in a list for example, we don't have to care about XSL.
Here is a little example I would use in my metadata sheet:
```

<code class="xml">
<span class="details">Formats disponibles: 
    <xsl:for-each select="./sdi:ExtendedMetadata/sdi:ex_Resource/sdi:ex_Metadata
/sdi:ex_Diffusionsdi:ex_ExtractionProperties/sdi:ex_ExtractionProperty[@name='Format']/sdi:ex_PropertyValue">
        <xsl:value-of select="./@label" />
        <xsl:if test="position() != last()" >
            <xsl:text>,  </xsl:text>
        </xsl:if>
    </xsl:for-each>
</span>

```


+For point 2: User waring :+
Yes, it may affect XSL work, but many other thing can affect it: all the model (class, attributes, values, relations) + diffusion, preview etc... Should we put warning everywhere? I don't think so.

+Magoni Bruno  wrote:+
> Such points are also applicable to already existing extended content such like ResourceType's name, Organism's name, Parent resource's name

Yes, you're right, here is the current status of those elements:

- ResourceType: label is actually not embedded in extendedMetadata (only name), but it's a good idea to have it
- Organism's name:  is not multilingual
- Parent resource's name (title): is already translated in extendedMetadata

+Last point about ISO19115 compliance and proxy+

I'm not aware of ISO 19115 Metadata extension information package logic, but is smells like a lot of work ;-)
I dont see what the interest would be... 
- The sdi extension we store in OGC catalog (GeoNetwork) is only technical, that may be filtered out... (but we can do this by xsl right?)
- The extendedMetadata part is generated on the fly (and not published by proxy), with data from joomla DB, so the proxy would have to do this job? and what for? A non standard part that has no real point outside easySDI?

**Comment by yblatti** [comment]
**Original author :** Blatti Yves

I propose to also add ResourceType label (translated)

**Comment by yblatti** [comment]
**Original author :** Magoni Bruno

After discussion with Yves, seems that this feature is sufficiently described to vote on it.
Summary is:
* diffusion properties, which are not currently integrated in metadata extensions, are added as described by Yves
* resource type's label is also added keeping same logical to bring all translated item in metadata extensions

**Comment by yblatti** [comment]
**Original author :** Mérour Xavier

+1 for me.

**Comment by yblatti** [comment]
**Original author :** Magoni Bruno

+1

**Comment by yblatti** [comment]
**Original author :** Teixeira Jérôme

OK +1

**Comment by yblatti** [comment]
**Original author :** Plainecassagne Laurent

+1

**Comment by yblatti** [comment]
**Original author :** Blatti Yves

Merged in 4.3.x
